Dockerfile-dev

```diff
@@ -1,24 +1,15 @@
-FROM python:3.6
+# This file builds a bigchaindb development
+# image and expects 'bigchaindb:latest' image
+# to be built from main 'Dockerfile' in this repo.
+# By building on top of the base image we avoid re-installing
+# all dependecies and doing things like 'apt-get update'
+# for each development build, this significantly speeds up
+# the build time in a dev environment.
+
+FROM bigchaindb:latest
 LABEL maintainer "dev@bigchaindb.com"
-
-RUN apt-get update \
-    && apt-get install -y vim \
-    && pip install -U pip \
-    && pip install pynacl \
-    && apt-get autoremove \
-    && apt-get clean
-
-ENV BIGCHAINDB_SERVER_BIND 0.0.0.0:9984
-ENV BIGCHAINDB_WSSERVER_HOST 0.0.0.0
-ENV BIGCHAINDB_WSSERVER_SCHEME ws
-
-ENV BIGCHAINDB_WSSERVER_ADVERTISED_HOST 0.0.0.0
-ENV BIGCHAINDB_WSSERVER_ADVERTISED_SCHEME ws
-
-ARG backend
-
 RUN mkdir -p /usr/src/app
 COPY . /usr/src/app/
 WORKDIR /usr/src/app
-RUN pip install --no-cache-dir -e .[dev]
-RUN bigchaindb -y configure "$backend"
+
+RUN pip install --no-cache-dir .
```

Comment:
Why was  `-e .[dev]` removed?

Reply:
The `-e` stands for `editable`. It is similar to `python setup.py develop`:

```bash
$ pip install --help | grep '\-\-editable'
  -e, --editable <path/url>   Install a project in editable mode (i.e. setuptools "develop mode") from a local project path 
```

We need when developing as it allows us to do away with the need of re-installing the project when files are changed.

Thus, the editable mode combined with the mounted volumes in docker-compose allow us to make changes and test them very quickly, without the need to execute `pip install` each time.